tests/test_all_rules.py

```diff
@@ -957,86 +956,34 @@ def test_event_override(self):
-            },
-            'eql_sq': {
-                'errors': [],
-                'msg': 'cannot have the `timestamp_override` set to `event.ingested` because it uses a sequence'
-            },
-            'lt_82_eql': {
-                'errors': [],
-                'msg': 'should have the `timestamp_override` set to `event.ingested`'
-            },
-            'lt_82_eql_beats': {
-                'errors': [],
-                'msg': ('eql rules include beats indexes. Non-elastic-agent indexes do not add the `event.ingested` '
-                        'field and there is no default fallback to @timestamp for EQL rules <8.2, so the override '
-                        'should be removed or a config entry included to manually add it in a custom pipeline')
-            },
-            'gte_82_eql': {
-                'errors': [],
-                'msg': ('should have the `timestamp_override` set to `event.ingested` - default fallback to '
-                        '@timestamp was added in 8.2')
-            }
-        }
-
-        pipeline_config = ('If enabling an EQL rule on a non-elastic-agent index (such as beats) for versions '
-                           '<8.2, events will not define `event.ingested` and default fallback for EQL rules '
-                           'was not added until 8.2, so you will need to add a custom pipeline to populate '
-                           '`event.ingested` to @timestamp for this rule to work.')
+        errors = []
 
         for rule in self.all_rules:
-            if rule.contents.data.type not in ('eql', 'query'):
-                continue
-            if rule.contents.metadata.get('integration'):
-                integrations = rule.contents.metadata.get('integration')
-                if not isinstance(integrations, list):
-                    integrations = [integrations]
-                machine_learning_packages_lower = [pkg.lower() for pkg in definitions.MACHINE_LEARNING_PACKAGES]
-                if any(tag in machine_learning_packages_lower for tag in integrations):
-                    continue
-            if isinstance(rule.contents.data, QueryRuleData) and 'endgame-*' in rule.contents.data.index:
-                continue
-
-            has_event_ingested = rule.contents.data.timestamp_override == 'event.ingested'
-            indexes = rule.contents.data.get('index', [])
-            beats_indexes = parse_beats_from_index(indexes)
-            min_stack_is_less_than_82 = Version.parse(rule.contents.metadata.min_stack_version or '7.13.0',
-                                                      optional_minor_and_patch=True) < Version.parse("8.2.0")
-            config = rule.contents.data.get('note') or ''
-            rule_str = self.rule_str(rule, trailer=None)
+            # skip ML rules, non-EQL or KQL rules, and rules with advanced analytic packages
+            if hasattr(rule.contents.data, 'language'):
```

Comment:
This is every rule type except for ML rules. Can we be more explicit here?

```suggestion
            if isinstance(rule.contents.data, QueryRuleData):
```

Reply:
Focusing on `QueryRuleData` works since it would be filtered on `definitions.FilterLanguages`.